src/main/java/com/epam/esm/dao/GiftCertificateDAO.java

```diff
@@ -0,0 +1,81 @@
+package com.epam.esm.dao;
+
+import com.epam.esm.dao.request.CertificateSearchCriteria;
+import com.epam.esm.model.GiftCertificate;
+import com.epam.esm.service.exception.ServiceException;
+
+import javax.persistence.PersistenceException;
+import java.util.List;
+
+/**
+ * This interface provides with ability to
+ * transfer {@code GiftCertificate} in and out
+ * of data source.
+ *
+ * @author Ilya Morozov
+ */
+public interface GiftCertificateDAO {
+
+    /**
+     * Retrieves data of {@code GiftCertificate} from
+     * data source by it name
+     * which equals to {@code String name}.
+     *
+     * @param name certificate name.
+     * @return {@code GiftCertificate}.
+     */
+    GiftCertificate getGiftCertificateByName(String name);
+
+    /**
+     * Retrieves data of {@code GiftCertificate} from
+     * data source by it id
+     * which equals to {@code int certificateId}.
+     *
+     * @param certificateId certificate id.
+     * @return {@code Order}.
+     */
+    GiftCertificate getGiftCertificateById(int certificateId);
+
+    /**
+     * Retrieves certain number of {@code GiftCertificate} from data source.
+     *
+     * @param searchCriteria object containing search criteria.
+     * @param page page number of {@code GiftCertificate} to return.
+     * @param size page size of {@code GiftCertificate} to return from data source.
+     * @return List<GiftCertificate> - certain number of existing gift certificates in data source.
+     */
+    List<GiftCertificate> getGiftCertificatesByRequestBody(CertificateSearchCriteria searchCriteria, int page, int size);
+
+    /**
+     * Retrieves number of a last page from data source if every page
+     * contains certain number of {@code GiftCertificate}.
+     *
+     * @param size size of a page.
+     * @return number of a last page.
+     */
+    int getLastPage(int size);
```

Comment:
I'd prefer to have common class for this operation rather than introduce duplication. Also getLastPage is not appropriate operation for GiftCertificate resource